src/pscf/iterator/ImposedFieldsTmpl.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+* PSCF - Polymer Self-Consistent Field Theory
+*
+* Copyright 2016 - 2022, The Regents of the University of Minnesota
+* Distributed under the terms of the GNU General Public License.
+*/
+
+#include "ImposedFieldsTmpl.h"
+
+namespace Pscf {
+
+   // Constructor
+   ImposedFieldsTmpl::ImposedFieldsTmpl()
+    : fieldGenPtr1_(0),
+      fieldGenPtr2_(0),
+      type_()
```

Comment:
Consider setting type explicitly to None rather than leaving a value uninitialized.  Derived class constructors are called after the base class constructor, and so will redefine the value to appropriate specific value later in the process.

Reply:
I did not address this in my broader comment: this type_ variable is a string that is read from the parameter file, so it is best to start with it uninitialized. I think you might be thinking of the FieldGenerator class, which has a variable type_ that is an enum parameter. In that case, we initialize it with type None. 

It is potentially confusing to use the same name for these two member variables that are different types and have different purposes, so if you'd like me to rename one of them then I can do that.